### PR TITLE
Add ProductRepository, CategoryRepository, ProductService, CategoryService 

### PR DESCRIPTION
**Pull request by NancyHamdallah**

### CODE REVIEW

File: src/data-access/CategoryRepository.ts
Comment (yousefkmail) [inline src/data-access/CategoryRepository.ts:1]: CreateCategory, findById, findAll, already exist in the base RepositoryBase, you don't need to implement them again.

File: src/data-access/Interfaces/ICategoryRepository.ts
Comment (yousefkmail) [inline src/data-access/Interfaces/ICategoryRepository.ts:1]: Here as well, you need to remove FindById, findAll and createCategory.

File: src/data-access/Interfaces/IProductRepository.ts
Comment (yousefkmail) [inline src/data-access/Interfaces/IProductRepository.ts:1]: Same here, you need to not implement methods again.

File: src/data-access/ProductRepository.ts
Comment (yousefkmail) [inline src/data-access/ProductRepository.ts:1]: Same here, remove the redundant methods, no need to do them again.

**Review by yousefkmail — CHANGES_REQUESTED** [review]
Please fix the issues on each file.

File: src/data-access/CategoryRepository.ts
Comment (yousefkmail) [inline src/data-access/CategoryRepository.ts:1]: CreateCategory, findById, findAll, already exist in the base RepositoryBase, you don't need to implement them again.

File: src/data-access/Interfaces/ICategoryRepository.ts
Comment (yousefkmail) [inline src/data-access/Interfaces/ICategoryRepository.ts:1]: Same here, Remove the corresponding methods.

File: src/data-access/Interfaces/IProductRepository.ts
Comment (yousefkmail) [inline src/data-access/Interfaces/IProductRepository.ts:1]: Same here, you need to not implement methods again.

**Review by yousefkmail — CHANGES_REQUESTED** [review]

File: src/data-access/CategoryRepository.ts
Comment (NancyHamdallah) [reply src/data-access/CategoryRepository.ts:1]: Done

File: src/data-access/Interfaces/ICategoryRepository.ts
Comment (NancyHamdallah) [reply src/data-access/Interfaces/ICategoryRepository.ts:1]: Done

File: src/data-access/Interfaces/IProductRepository.ts
Comment (NancyHamdallah) [reply src/data-access/Interfaces/IProductRepository.ts:1]: Done

**Review by yousefkmail — APPROVED** [review]

**Comment by yousefkmail** [comment]
Approved, you can merge.

**Pull request merged**